src/main.qml

```diff
@@ -302,11 +302,11 @@ ApplicationWindow {
                                 return
                             }
 
-                            if (!optionspopup.visible && imageWriter.imageSupportsCustomization()) {
-                                usesavedsettingspopup.openPopup()
-                            } else {
-                                confirmwritepopup.askForConfirmation()
-                            }
+                            if (!optionspopup.visible) {
+                                optionspopup.openPopup()
+                            } 
+                            confirmwritepopup.askForConfirmation()
```

Comment:
Is this going to show the confirm popup at the exact same time, since there is no else statement here anymore?

Reply:
Hmmm I might need to think about how to re-write this. The way it's currently implemented, the OptionsPopup actually inherits from Window instead of from Popup, so it isn't modal and can currently be opened completely on its own at any time with the Ctrl + Shift + X shortcut. Before, the flow of events was UseSavedSettingsPopup (which blocks the rest of the window, because it does actually inherit from Popup and modal is set to true) -> click edit settings button -> display OptionsPopup -> click either X or Save -> return to UseSavedSettingsPopup, which is still blocking off access to the main window.

Do you have any strong feelings about me removing the Ctrl + Shift + X shortcut and making OptionsPopup an actual modal popup? If you'd rather not, I might be able to fudge similar behavior by modifying the on save clicked/on exit functions for the OptionsPopup instead

Reply:
It would be really smart to turn it into a modal. Ideally it would just pop up inside the app at some point (rather than separate) - so this change makes the most sense to me, just make a part of the steps to create a USB every time.

Reply:
After testing out making the popup an actual Popup, I can see why they went with Window instead - apparently Popups can't be any larger than the parent window. However, I did find that Window has a modality attribute that can be set to mimic the blocking effect, and then I changed the logic a little so that when the user clicks next, the options popup is displayed, then if the user clicks save on the options popup, it closes the options popup and sends a signal to the main window to display the confirm write popup.

![optionspopup](https://github.com/unraid/usb-creator-next/assets/47037553/aae06f15-f0af-42d0-89c3-d35bd82c9437)